Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp

```diff
@@ -99,12 +99,27 @@ WebIDL::ExceptionOr<void> CSSStyleSheet::remove_rule(unsigned index)
 
 void CSSStyleSheet::for_each_effective_style_rule(Function<void(CSSStyleRule const&)> const& callback) const
 {
+    for (auto& media : media()) {
+        if (!media.matches()) {
+            return;
+        }
+    }
+
     m_rules->for_each_effective_style_rule(callback);
 }
 
 bool CSSStyleSheet::evaluate_media_queries(HTML::Window const& window)
 {
-    return m_rules->evaluate_media_queries(window);
+    bool any_media_queries_changed_match_state = false;
+    for (auto& media : m_media_list) {
+        if (media.evaluate(window))
+            any_media_queries_changed_match_state = true;
```

Comment:
This doesn't look right. As the name indicates, `any_media_queries_changed_match_state` should be `true` if one or more media queries have *changed* evaluation state.
With this change, we will not notice a transition from `true` to `false` in evaluation state.
Look at `CSSRuleList::evaluate_media_queries()` for how the before/after values are calculated.